warehouse/query-core/src/main/java/datawave/query/iterator/FieldIndexOnlyQueryIterator.java

```diff
@@ -253,7 +253,18 @@ protected void createAndSeekIndexIterator(Range range, Collection<ByteSequence>
         Collection<String> unindexedTypes = Lists.newArrayList();
         
         Set<String> keys = fetchDatatypeKeys(this.documentOptions.get(NON_INDEXED_DATATYPES));
-        unindexedTypes.addAll(keys);
+        
+        String compressedOptionString = this.documentOptions.get(QUERY_MAPPING_COMPRESS);
+        if (compressedOptionString != null) {
```

Comment:
Replace this check with StringUtils.isBlank(). That will catch the case when QUERY_MAPPING_COMPRESS is set to an empty string.